src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
I'm not saying there is any accidental quadratic behaviour in that code. It's never going to be "noticeable" on its own when you're running the full compiler. But it's doing more work and creating more garbage that we don't need to do except for when generating documentation.

This is also a case for separation of concerns. Ide will want to run the desugaring pipeline decoupled from parser and type checker to do name resolution (It doesn't care for the ordering of exports), and when the user is directly interacting with it milliseconds do degrade the user experience.